packages/core/src/eventTarget.ts

```diff
@@ -16,6 +16,9 @@ class CornerstoneEventTarget implements EventTarget {
     if (!this.listeners[type]) {
       this.listeners[type] = [];
     }
+    if (this.listeners[type].indexOf(callback) !== -1) {
```

Comment:
This is a standard event add check to prevent multiple callbacks - makes debugging easier, and for large viewport layouts, is faster and appears to reduce the occurance of a browser crash.